src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -83,20 +93,52 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
             // Description
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
```

Comment:
Well, refactoring in general is :ok: just not reordering / regrouping. Fine for example if you want to create a short method containing these few lines for the 3 properties needing similar things.